src/HttpMethodsClient.php

```diff
@@ -11,6 +11,14 @@
  *
  * Use this interface when you do not have PSR-7 RequestInterface instances available.
  *
+ * Implementations should be configured in their constructor. For runtime behaviour changes, there
+ * is an $options parameter. It SHOULD be validated by the client and an exception thrown if
+ * unknown options are passed.
```

Comment:
>     exception thrown if unknown options are passed.
> 
> How to handle specific options for an adapter then ?

only pass them when you know you are using that specific client/adapter.
in a library, such options - if needed at all - should be injected.

to be honest i am a bit unsure about this. it would be cleaner to not
have options at all. but it could lead to an excessive number of
adapter/client instances to have them with all the various options.
maybe we should still do it that way.